app/src/main/java/fr/neamar/kiss/MainActivity.java

```diff
@@ -305,9 +305,18 @@ protected void onDestroy() {
         super.onDestroy();
         // unregister our receiver
         this.unregisterReceiver(this.mReceiver);
+        KissApplication.getCameraHandler().releaseCamera();
     }
+    
+    
 
     @Override
+	protected void onPause() {		
+		super.onPause();
+		KissApplication.getCameraHandler().releaseCamera();
```

Comment:
onPause() is called very often, so if we can avoid getting an empty handle to release it immediately too ;)